### PR TITLE
MAKE-821: add functions to find/delete users by name

### PR DESCRIPTION
**Pull request by Kimchelly**
JIRA: https://jira.mongodb.org/browse/MAKE-821

Since the depot only has operations on individual document fields based on tags, this adds functionality to operation on entire documents stored in the database.

### CODE REVIEW

**Comment by julianedwards** [comment]
I have a [ticket](https://jira.mongodb.org/browse/MAKE-822) in my sprint to create wrapper functions for convenience functions in certstrap that do not require using the "tag" extension... see [here](https://github.com/square/certstrap/blob/master/depot/pkix.go). Also, I do not think that it is a good idea to create public functions on mongoDepot that are not part of the Depot interface. Also, I have a delete on expiration function, which can easily be refactored into a regular delete operation, that removes entire documents.

File: db.go [hunk 1]
```diff
@@ -53,3 +53,32 @@ func (m *mongoDepot) DeleteExpiresBefore(cutoff time.Time) error {
 func expiresBeforeQuery(cutoff time.Time) bson.M {
 	return bson.M{userTTLKey: bson.M{"$lte": cutoff}}
 }
+
+// FindUser finds the User in the database by name.
+func (m *mongoDepot) FindUser(name string) (*User, error) {
```
Comment (tychoish) [inline db.go]: do we need to change the interface to make these more broadly accessible?

Comment (julianedwards) [reply db.go]: No, see this [ticket](https://jira.mongodb.org/browse/MAKE-822)

**Comment by Kimchelly** [comment]
Closing this until [this ticket](https://jira.mongodb.org/browse/MAKE-822) is done. Ideally, I would like to be able to insert/update/delete documents atomically rather than do operations on a field-by-field basis.

**Pull request closed without merging**